### PR TITLE
Find SceneObject by name #239

### PR DESCRIPTION
**Pull request by liaxim**
Is this what @chenchao1407 had in mind?

### CODE REVIEW

File: GVRf/Framework/src/org/gearvrf/GVRSceneObject.java [hunk 1]
```diff
@@ -563,6 +563,23 @@ public void removeChildObject(GVRSceneObject child) {
...
+     * Performs case-sensitive search
+     * 
+     * @param name
+     * @return null if nothing was found or name was null
+     */
+    public GVRSceneObject[] getSceneObjectsByName(final String name) {
+        if (null == name) {
+            return null;
+        }
+
```
Comment (thomasflynn) [inline GVRf/Framework/src/org/gearvrf/GVRSceneObject.java]: maybe i'm missing it, what if this object's name matches?

**Comment by thomasflynn** [comment]
yeah, off-hand, this looks like it'll do it.  i only have the one question in the code.  @salgubel, i know you guys wrote your own version of this.  will this meet your needs so you don't need to use your own code for this?

File: GVRf/Framework/src/org/gearvrf/GVRSceneObject.java (hunk 1 above)
Comment (liaxim) [inline GVRf/Framework/src/org/gearvrf/GVRSceneObject.java]: Good point, will upload a fix.

**Comment by liaxim** [comment]
Checking the current GVRSceneObject's name and in addition to null queries, excluding empty string queries.

File: GVRf/Framework/src/org/gearvrf/GVRSceneObject.java [hunk 2]
```diff
@@ -563,6 +563,26 @@ public void removeChildObject(GVRSceneObject child) {
...
 
     /**
+     * Performs case-sensitive search
+     * 
+     * @param name
+     * @return null if nothing was found or name was null/empty
+     */
+    public GVRSceneObject[] getSceneObjectsByName(final String name) {
+        if (null == name || name.isEmpty()) {
+            return null;
```
Comment (lalwani) [inline GVRf/Framework/src/org/gearvrf/GVRSceneObject.java]: Another small fail-safe check might be to do name.trim().isEmpty().

**Comment by lalwani** [comment]
Looks good

File: GVRf/Framework/src/org/gearvrf/GVRSceneObject.java (hunk 2 above)
Comment (Liahim) [inline GVRf/Framework/src/org/gearvrf/GVRSceneObject.java]: My goal was to filter out the name assigned by default (empty string). null could be considered an alias for this from user's perspective. Didn't want this API to be easily used as a replacement of getWholeSceneObjects.

If the user wants to use various number of spaces as names, I say let's not second guess the user.

Comment (lalwani) [inline GVRf/Framework/src/org/gearvrf/GVRSceneObject.java]: makes sense.

**Comment by liaxim** [comment]
jMonkeyEngine's getChild(name) returns the first matching node.

**Comment by liaxim** [comment]
Doesn't seem like OpenSceneGraph has direct match for this api; per http://trac.openscenegraph.org/projects/osg//wiki/Support/Tutorials/FindingNodes it is expected for the users to roll their own implementations.

**Comment by thomasflynn** [comment]
perhaps we can also add a getSceneObjectByName (singular rather than plural) that returns the first matching node.  i'll merge this in the meantime.

**Pull request merged**